src/dbnode/storage/database.go

```diff
@@ -889,40 +895,130 @@ func (d *db) ReadEncoded(
 			xopentracing.Time("end", end),
 		)
 	}
+
 	defer sp.Finish()
+	return n.ReadEncoded(ctx, id, start, end)
+}
 
+func (d *db) WideQuery(
+	ctx context.Context,
+	namespace ident.ID,
+	query index.Query,
+	start time.Time,
+	shards []uint32,
+	iterOpts index.IterationOptions,
+) ([]ident.IndexChecksum, error) { // FIXME: change when exact type known.
```

Comment:
nit: can remove comment now, ya?

Reply:
Unfortunately not... this will still change in future once more downstream logic is added (this will be the entry point into entire wide query flow, including later steps); this is a temporary return type to make integration testing easier